### PR TITLE
Delete comment

### PR DESCRIPTION
**Pull request by erinlimbo**
Add ability to delete comments and memes

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/Delete.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  /** Delete the element of the given key id */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String reader = request.getReader().lines().collect(Collectors.joining());
+    JsonObject deleteObj = new JsonParser().parse(reader).getAsJsonObject();
+    long id = Long.parseLong(deleteObj.get("id").getAsString());
+    String type = deleteObj.get("type").getAsString();
+
+    Query query = new Query(type).addSort("timeStamp", SortDirection.DESCENDING);
```
Comment (laskdaf) [inline portfolio/src/main/java/com/google/sps/servlets/Delete.java]: seems like `query` isn't being used

File: portfolio/src/main/java/com/google/sps/servlets/Delete.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Key deleteKey = KeyFactory.createKey(type, id);
```
Comment (laskdaf) [inline portfolio/src/main/java/com/google/sps/servlets/Delete.java:36]: I'm not familiar with how keys are determined in the datastoreservice, but is it that this gives you the key of the entry of given type and id? 

also if each id is unique, why do you need to specify the type? 
is it not enough to use `entity.getKey()`?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -135,23 +137,51 @@ async function loadMemes() {
...
 
         memeDiv.classList.add("meme-div");
         memeContainer.appendChild(memeDiv);
 
     })
 }
 
+/** Delete this element. */
+async function deleteThis(elementType, elementId) {
+    let response = await fetch('/delete', {
```
Comment (laskdaf) [inline portfolio/src/main/webapp/script.js]: if you're not using response, you don't need to set it to a variable

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -135,23 +137,51 @@ async function loadMemes() {
...
+        body: JSON.stringify({
+            type: elementType, 
+            id: elementId
+        }),
+    });
+}
+
+/** Delete all comments from the page and datastore. */
+async function deleteComments() {
+    let response = await fetch('/delete-data', {
```
Comment (laskdaf) [inline portfolio/src/main/webapp/script.js]: const

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -135,23 +137,51 @@ async function loadMemes() {
...
+        method: 'POST',
+    });
+    response.then(getComments());
```
Comment (laskdaf) [inline portfolio/src/main/webapp/script.js]: why are you waiting for response to resolve before calling `getComments()`

i think you should be able to just call `getComments()` like before. What changed?

Comment (erinlimbo) [reply portfolio/src/main/webapp/script.js]: Oh idk why this wasnt deleted, I meant to delete this, but anyways I was trying to make the html update instantly. I now realize that the code only relates to the backend and that I need to actually change the html.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (erinlimbo) [reply portfolio/src/main/webapp/script.js]: gotcha

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (erinlimbo) [reply portfolio/src/main/webapp/script.js]: removing variable

File: portfolio/src/main/java/com/google/sps/servlets/Delete.java (hunk 2 above)
Comment (erinlimbo) [reply portfolio/src/main/java/com/google/sps/servlets/Delete.java:36]: I also didnt know how it worked much so i went onto Cupcake for help.
I could try tesing out taking out the type and see if the unique ID will still work.

Comment (erinlimbo) [reply portfolio/src/main/java/com/google/sps/servlets/Delete.java:36]: yea it doesnt work without it I get a compile error, im not sure how Keys work much either

Comment (laskdaf) [reply portfolio/src/main/java/com/google/sps/servlets/Delete.java:36]: ok if it's functionally correct then resolve this and i'll approve it.

**Review by laskdaf — APPROVED** [review]

**Pull request merged**